core/kazoo_voicemail/src/kvm_message.erl

```diff
@@ -87,13 +84,60 @@ new(Call, Options) ->
             lager:debug("storing voicemail media recording ~s in doc ~s", [AttachmentName, MessageId]),
             case store_recording(AttachmentName, MediaUrl, kapps_call:exec(Funs, Call), MessageId) of
                 'ok' ->
-                    notify_and_update_meta(Call, MessageId, Length, Options);
+                    maybe_convert_attachment(Call, MessageId, Length, Options);
                 {'error', Call1} ->
+                    Msg = io_lib:format("failed to store voicemail media ~s in voicemail box ~s of account ~s"
+                                       ,[MessageId, BoxId, kapps_call:account_id(Call)]
+                                       ),
                     lager:error(Msg),
                     {'error', Call1, Msg}
             end
     end.
 
+-spec maybe_convert_attachment(kapps_call:call(), kz_term:ne_binary(), integer(), kz_term:proplist()) -> {'ok', kapps_call:call()}.
+maybe_convert_attachment(Call, MediaId, Length, Options) ->
+    AccountId = kapps_call:account_id(Call),
+    Db = kvm_util:get_db(AccountId, MediaId),
+    {'ok', MediaDoc} = kz_datamgr:open_doc(Db, MediaId),
+    case kzd_box_message:transcribe_filename(MediaDoc) =/= 'undefined'
+        andalso convert_attachment(Db, MediaDoc) of
```

Comment:
Trying to move away from returning non-booleans on the right side of `andalso` expressions. I would prefer to see a function call to handle the clauses:

```erlang
    {'ok', MediaDoc} = kz_datamgr:open_cache_doc(Db, MediaId),
    maybe_convert_attachment(Call, MediaId, Length, Options, MediaDoc, kzd_box_message:transribe_filename(MediaDoc)).

maybe_convert_attachment(Call, MediaId, Length, Options, _MediaDoc, 'undefined') ->
    notify_and_update_meta(...);
maybe_convert_attachment(Call, MediaId, Length, Options, _MediaDoc, _Filename) ->
    convert_attachment(Call, MediaId, Length, Options, MediaDoc).

convert_attachment(Call, MediaId, Length, Options, MediaDoc) ->
    Db = kvm_util:get_db(kapps_call:account_id(Call), MediaId),
    Conversion = convert_attachment(Db, MediaDoc),
    handle_conversion(Call, MediaId, ...)
```

Anyway, hopefully that gets the idea across. Now you have a function for explicitly doing the conversion while the `maybe_` function doesn't do anything apart from testing conditions for conversion.